holoviews/plotting/bokeh/element.py

```diff
@@ -295,6 +305,7 @@ def _init_tools(self, element, callbacks=None):
         tooltips, hover_opts = self._hover_opts(element)
         tooltips = [(ttp.pprint_label, '@{%s}' % util.dimension_sanitizer(ttp.name))
                     if isinstance(ttp, Dimension) else ttp for ttp in tooltips]
+        tooltips += self._group_label(element)
```

Comment:
not sure I understand this part. Aren't I already doing the check in the method whether the data contains `label` or `group`? Are you suggesting that the method should return an unaltered tooltips list rather than an empty list, if `label` or `group` are not defined by the user?

Reply:
If the data has a `label` or `group` dimension, that should take precedence over the elements. An example is the following:

``` python
import holoviews as hv
import pandas as pd
hv.extension("bokeh")

hv.Curve(pd.DataFrame(dict(label=range(4)))).opts(tools=['hover'])
```

Reply:
ah ok I understand now, thanks for clarifying

Reply:
since the tooltip dims are already sanitized, I think this can simply be done by passing in tooltips as you say and then adding this to each conditional:

`'label' not in [t[0] for t in tooltips]`

Reply:
<details> <summary> Code </summary>

```python
import numpy as np
import panel as pn
import pandas as pd
import holoviews as hv
hv.extension('bokeh')

x = np.linspace(0, 10*np.pi)
curves = {}
j = 0.5
yvals = j * np.sin(x)
curves = []
curves.append(hv.Curve((x, yvals)).opts(tools=['hover'], color='blue'))
curves.append(hv.Curve((x + 2 * np.pi/2, 2 * yvals), label='curve2').opts(tools=['hover'], color='red'))
curves.append(hv.Curve((x + 3 * np.pi/2, 3 * yvals), group='A').opts(tools=['hover'], color='green'))
curves.append(hv.Curve((x + 4 * np.pi/2, 4 * yvals), label='curve4', group='A').opts(tools=['hover'], color='orange'))
curves.append(hv.Curve(pd.DataFrame(dict(label=range(4))), label='no_label_ttip').opts(tools=['hover'], color='purple'))
pn.Row(hv.Overlay(curves).opts(width=600, height=600)).servable()
```

</details>


https://github.com/holoviz/holoviews/assets/6613202/87001830-3da0-4641-9c4b-e47cad173534